edx_exams/apps/core/management/commands/bulk_add_course_staff.py

```diff
@@ -0,0 +1,98 @@
+"""Management command to populate the CourseStaffRole model and related User objects, from LMS, using CSV"""
+import logging
+import time
+
+import unicodecsv
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from edx_exams.apps.core.models import CourseStaffRole, User
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Management command to add Course Staff (and User) in batches from CSV
+    """
+    help = """
+    Add Course Staff in bulk from CSV.
+    Expects that the data will be provided in a csv file format with the first row
+    being the header and columns being: username, email, role, course_id.
+    Example:
+        $ ... bulk_add_course_staff --csv_path=foo.csv
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-p', '--csv_path',
+            metavar='csv_path',
+            dest='csv_path',
+            required=False,
+            help='Path to CSV file.')
+        parser.add_argument(
+            '--batch_size',
+            type=int,
+            default=200,
+            dest='batch_size',
+            help='Batch size')
+        parser.add_argument(
+            '--batch_delay',
+            type=float,
+            default=1.0,
+            dest='batch_delay',
+            help='Time delay in each iteration')
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+        """
+        The main logic and entry point of the management command
+        """
+        csv_path = options['csv_path']
+        batch_size = options['batch_size']
+        batch_delay = options['batch_delay']
+
+        if csv_path:
+            with open(csv_path, 'rb') as csv_file:
+                self.add_course_staff_from_csv(csv_file, batch_size, batch_delay)
+
+        logger.info('Bulk add course staff complete!')
+
+    def add_course_staff_from_csv(self, csv_file, batch_size, batch_delay):
+        """
+        Add the given set of course staff provided in csv
+        """
+        reader = list(unicodecsv.DictReader(csv_file))
```

Comment:
What made you choose the `unicodecsv` package? I ask because Python's native `csv` module has a `DictReader` class that should provide the same functionality. Looking at the docs for `unicodecsv`, it looks like it improves upon some unicode issues in Python 2, but we're on Python 3 here, so I'd imagine we don't benefit from `unicodecsv`. Where possible, I think it would be better to leverage native packages. I think this should be an easy swap without any impact to the command or its tests. What do you think?

As a nit, at least for the `csv.DictReader` class, the class is iterable, so you don't have to cast the value to a list; you can just iterate the instance. Maybe `unicodecsv.DictReader` is different? Oh, I see. You're casting it to a list because you're iterating `reader` twice and don't want to exhaust the iterable, right?

Reply:
Ah, good point! I've changed to use csv. The only reason I used unicode was because I had referenced code from elsewhere... so not really a good reason haha. 

I primarily cast to a list because I check the length of the reader (in order to batch the creates for the course staff). Lemme know if this is a bad idea though!